shared/settings/advanced/index.js

```diff
@@ -33,11 +34,47 @@ const Advanced = (props: Props) => (
         />
       </Box>
     )}
-    <DBNuke {...props} />
+    <Developer {...props} />
   </Box>
 )
 
-function DBNuke(props: Props) {
+type TraceButtonProps = {
+  durationSeconds: number,
+  onTrace: (durationSeconds: number) => void,
+}
+
+type TraceButtonState = {
+  enabled: boolean,
+}
+
+class TraceButton extends React.Component<TraceButtonProps, TraceButtonState> {
+  state = {
+    enabled: true,
+  }
+
+  _onClick = () => {
+    this.setState({enabled: false})
+    this.props.onTrace(this.props.durationSeconds)
+    setTimeout(() => {
+      this.setState({enabled: true})
```

Comment:
this is not safe. use the timerhoc to wrap this component

Reply:
Changed to do the delay in the saga and to store the state in state.waiting, so that it survives tab changes